### PR TITLE
Fix unit tests to follow style used in minio

### PR DESCRIPTION
**Pull request by balamurugana**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [ ] All new and existing tests passed.

### CODE REVIEW

File: pkg/http/server_test.go [hunk 1]
```diff
@@ -98,79 +99,85 @@ func TestNewServer(t *testing.T) {
...
+			addr := "127.0.0.1:" + getNextPort()
+
+			server := NewServer([]string{addr},
+				http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+					fmt.Fprintf(w, "Hello, world")
+				}),
+				&certificate)
+			if testCase.resetServerCiphers {
+				// Use Go default ciphers.
+				server.TLSConfig.CipherSuites = nil
```
Comment (aead) [inline pkg/http/server_test.go:152]: Can you explain what you're trying to achieve with this?
Handling `TLSConfig.CipherSuites` correctly is the responsibility of the standard library. Testing the standard library is IMO not useful. So setting `TLSConfig.CipherSuites` to an arbitrary value should not change anything except that a client has to support at least one of those cipher suites.

 So there are two important cases:
1. A client who only supports "unsecure" ciphers should not be able to connect
2. A client who supports at least one "secure" cipher should be able to connect
(3. A default Go client should be able to connect for backward comparability)

What is the point of testing whether a default Go client can connect to a server with the default cipher suites?

**Review by aead — CHANGES_REQUESTED** [review]

File: pkg/http/server_test.go (hunk 1 above)
Comment (balamurugana) [reply pkg/http/server_test.go:152]: Could you list down what test cases are valid so that I can remove rest of them?

Comment (aead) [reply pkg/http/server_test.go:152]: - [That](https://github.com/aead/minio/pull/3/files#diff-e3962e3c7025eecdb6dc51672e35b223R135) would be the backward compatibility test (default Go client can connect)
- [That](https://github.com/aead/minio/pull/3/files#diff-e3962e3c7025eecdb6dc51672e35b223R136) would test whether a client which supports (at least one) cipher suites of the server can connect
- [That](https://github.com/aead/minio/pull/3/files#diff-e3962e3c7025eecdb6dc51672e35b223R137) would test that the server rejects clients who does not support secure ciphers - and following that the server does not support "insecure" ciphers.

The other 3 are not really useful IMO - Or am I missing your point?

Comment (balamurugana) [reply pkg/http/server_test.go:152]: To cover with your current test code, I am going to retain https://github.com/aead/minio/pull/3/files#diff-e3962e3c7025eecdb6dc51672e35b223R138 and remove other two

Comment (aead) [reply pkg/http/server_test.go:152]: Okay, lets agree on that 👍

**Comment by aead** [comment]
@balamurugana CI is failing because of data race (race detector)

**Comment by balamurugana** [comment]
@aead Fixed

**Pull request merged**